tests/integration_tests/performance/test_snapshot_perf.py

```diff
@@ -247,180 +238,68 @@ def snapshot_resume_producer(logger, vm_builder, snapshot, snapshot_type, use_ra
-    io_engine = context.custom["io_engine"]
-    diff_snapshots = snapshot_type == SnapshotType.DIFF
+    With each previous firecracker version, create a snapshot and try to
+    restore in current version.
 
-    logger.info(
-        """Measuring snapshot resume({}) latency for microvm: \"{}\",
-    kernel {}, disk {} """.format(
-            snapshot_type,
-            context.microvm.name(),
-            context.kernel.name(),
-            context.disk.name(),
-        )
-    )
+    @type: performance
+    """
+    logger = logging.getLogger("old_snapshot_load")
+    snapshot_type = SnapshotType.FULL
+    jailer = firecracker_release.jailer()
+    fc_version = firecracker_release.base_name()[1:]
+    logger.info("Firecracker version: %s", fc_version)
+    logger.info("Source Firecracker: %s", firecracker_release.local_path())
+    logger.info("Source Jailer: %s", jailer.local_path())
 
-    # Create a rw copy artifact.
-    rw_disk = context.disk.copy()
-    # Get ssh key from read-only artifact.
-    ssh_key = context.disk.ssh_key()
-    # Create a fresh microvm from artifacts.
-    vm_instance = vm_builder.build(
-        kernel=context.kernel,
-        disks=[rw_disk],
-        ssh_key=ssh_key,
-        config=context.microvm,
-        diff_snapshots=diff_snapshots,
-        use_ramdisk=True,
-        io_engine=io_engine,
-    )
-    basevm = vm_instance.vm
-    basevm.start()
+    vcpus, guest_mem_mib = 2, 512
+    microvm_cfg = f"{vcpus}vcpu_{guest_mem_mib}mb.json"
+    vm = microvm_factory.build(guest_kernel, rootfs, monitor_memory=False)
+    vm.spawn()
+    vm.basic_config(vcpu_count=vcpus, mem_size_mib=guest_mem_mib)
+    iface = NetIfaceConfig()
+    vm.add_net_iface(iface)
+    vm.start()
 
     # Check if guest works.
-    exit_code, _, _ = basevm.ssh.execute_command("ls")
+    exit_code, _, _ = vm.ssh.execute_command("ls")
     assert exit_code == 0
 
-    logger.info("Create {}.".format(snapshot_type))
+    # The snapshot builder expects disks as paths, not artifacts.
+    disks = [vm.rootfs_file]
     # Create a snapshot builder from a microvm.
-    snapshot_builder = SnapshotBuilder(basevm)
-
+    snapshot_builder = SnapshotBuilder(vm)
     snapshot = snapshot_builder.create(
-        [rw_disk.local_path()], ssh_key, snapshot_type, use_ramdisk=True
+        disks, rootfs.ssh_key(), snapshot_type, net_ifaces=[iface]
     )
+    vm.kill()
 
-    basevm.kill()
-
-    st_core = core.Core(name="snapshot_resume_latency", iterations=SAMPLE_COUNT)
+    st_core.name = "older_snapshot_resume_latency"
+    st_core.iterations = SAMPLE_COUNT
 
     prod = producer.LambdaProducer(
         func=snapshot_resume_producer,
         func_kwargs={
             "logger": logger,
-            "vm_builder": vm_builder,
+            "vm_builder": MicrovmBuilder(bin_cloner_path),
             "snapshot": snapshot,
             "snapshot_type": snapshot_type,
-            "use_ramdisk": True,
+            "use_ramdisk": False,
```

Comment:
Any particular reason for changing the test to not use initrd?

Reply:
The snapshot restore path with a ramdisk is a bit bizarre and full of surprises TBH. We get the same results and the code is a bit simpler.